src/AIS.cpp

```diff
@@ -129,3 +131,107 @@ ais_base::VoyageInformation AIS::getVoyageInformation(ais::message_05 const& mes
     info.destination = message.get_destination();
     return info;
 }
+
+std::pair<Eigen::Quaterniond, ais_base::PositionCorrectionStatus> vesselToWorldOrientation(
+    const std::optional<base::Angle>& yaw,
+    const std::optional<base::Angle>& course_over_ground,
+    double speed_over_ground)
+{
+    if (yaw.has_value() && !std::isnan(yaw.value().getDeg())) {
```

Comment:
there is no reason for checking against NaN with degrees

```suggestion
    if (yaw.has_value() && !std::isnan(yaw.value().getRad())) {
```